$MirOS: ports/devel/glib/patches/patch-glib_gtimer_c,v 1.2 2009/03/02 20:12:13 bsiegert Exp $
--- glib/gtimer.c.orig	Fri Sep 12 19:55:28 2008
+++ glib/gtimer.c	Thu Nov 25 12:28:08 2010
@@ -402,6 +402,7 @@ gchar *
 g_time_val_to_iso8601 (GTimeVal *time_)
 {
   gchar *retval;
+  time_t real_tv_sec;
 #ifdef HAVE_GMTIME_R
   struct tm tm_;
 #endif
@@ -412,12 +413,14 @@ g_time_val_to_iso8601 (GTimeVal *time_)
 #define ISO_8601_FORMAT "%Y-%m-%dT%H:%M:%SZ"
   retval = g_new0 (gchar, ISO_8601_LEN + 1);
 
+  /* time_->tv_sec is of type glong, not time_t */
+  real_tv_sec = time_->tv_sec;
   strftime (retval, ISO_8601_LEN,
 	    ISO_8601_FORMAT,
 #ifdef HAVE_GMTIME_R
-	    gmtime_r (&(time_->tv_sec), &tm_)
+	    gmtime_r (&real_tv_sec, &tm_)
 #else
-	    gmtime (&(time_->tv_sec))
+	    gmtime (&real_tv_sec)
 #endif
             );
   
